databaseutil/databaseutil.go

```diff
@@ -94,6 +94,59 @@ func GetPasswordForUserWithEmailAddress(emailAddress string) ([]byte, error) {
 	return password, nil
 }
 
+func StoreNewNote(authorId int64, content string, creationTime time.Time) (int64, error) {
+	sqlQuery := `
+		INSERT INTO note (author_id, content, creation_time)
+		VALUES ($1, $2, $3)
+		RETURNING id`
+
+	rows, err := db.Query(sqlQuery, authorId, content, creationTime)
+	if err != nil {
+		return -1, convertPostgresError(err)
+	}
+	defer rows.Close()
+
+	var lastInsertId int64
+	for rows.Next() {
+
+		if lastInsertId != 0 {
+			return -1, QueryResultContainedMultipleRowsError
+		}
+
+		if err := rows.Scan(&lastInsertId); err != nil {
+			return -1, convertPostgresError(err)
+		}
+	}
+
+	if lastInsertId == 0 {
+		return -1, QueryResultContainedNoRowsError
+	}
+
+	if err := rows.Err(); err != nil {
+		return -1, convertPostgresError(err)
+	}
+
+	return lastInsertId, nil
+}
+
+func StoreNoteCategoryRelationship(noteId int64, category string) error {
+	sqlQuery := `
+		INSERT INTO note_to_category_relationship (note_id, type)
+		VALUES ($1, $2)`
+
+	rows, err := db.Query(sqlQuery, noteId, category)
+	if err != nil {
+		return convertPostgresError(err)
```

Comment:
we need to explicitly handle the case where `note_id` already exists in the table

Reply:
That should result in a database error, since an insert would make the note_id no longer unique In later branches i create a update note category relationship. That takes handles the existing note_id case: https://github.com/atmiguel/cerealnotes/blob/7a1a911bced216a0c74ae6bbe87f316970b12381/databaseutil/databaseutil.go#L181 If you want I we can only have a insertOrUpdate command, instead of two separate commands.

Reply:
Well my thought was that the front-end would either do a POST or a PUT. If they're doing a POST and the relationship already exists, I expect we'd return a 4xx. Conversely, if they're doing a PUT and the relationship does not exist, I expect we'd return a 4xx as well. If we just had an `insertOrUpdate` function we wouldn't be able to handle those cases

Reply:
Okay. Right now if you put an id that already exist it'll return as a 500 error. I'll add to technical debt to update `convertPostgresError` to handle duplicate key error.